packages/cyberstorm/src/components/Forms/CreateTeamForm/CreateTeamForm.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+import { faArrowsRotate } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useController, useForm, Form } from "react-hook-form";
+import * as Button from "../../Button";
+import { TextInput } from "../../TextInput/TextInput";
+import { useToast } from "../../Toast/Provider";
+import styles from "./CreateTeamForm.module.css";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { isErrorResponse } from "../../../utils/type_guards";
+import { useEffect } from "react";
+
+export function CreateTeamForm() {
+  const toast = useToast();
+
+  // "Required field" error is different from value being too little
+  // Check that the value is "> 0"
+  const schema = z.object({
+    teamName: z
+      .string({ required_error: "Team name is required" })
+      .min(1, { message: "Team name is required" }),
+  });
+
+  const {
+    control,
+    formState: { isSubmitting, errors },
+  } = useForm<z.infer<typeof schema>>({
+    mode: "onSubmit",
+    resolver: zodResolver(schema),
+  });
+
+  const teamName = useController({
+    control: control,
+    name: "teamName",
+  });
+
+  // We have to do this because the RHF Form component is in beta and it
+  // doesn't have a callback prop like "onValidation" that could take in
+  // the generalized addToast error informing block.
+  useEffect(() => {
+    if (errors.teamName) {
+      console.log(errors);
+      toast.addToast({
+        variant: "danger",
+        message: errors.teamName.message,
+        duration: 30000,
+      });
+    }
+  }, [errors]);
```

Comment:
I can't be sure but it's possible this objects gets recreated on each render and thus the effect also runs each time. Anyways the function would actually only need `errors.teamName?.message`, so just use it as the dependency. As a general rule prefer primitive values in the dependency array if feasible, that way you don't have to worry about this.

But I assume we don't really need the useEffect at all since I assume we don't want to show field errors on the toast component.